test/modules/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportGenerator.cs

```diff
@@ -21,31 +21,41 @@ sealed class DirectMethodReportGenerator : ITestResultReportGenerator
 
         internal DirectMethodReportGenerator(
             string trackingId,
-            string expectedSource,
-            ITestResultCollection<TestOperationResult> expectedTestResults,
-            string actualSource,
-            ITestResultCollection<TestOperationResult> actualTestResults,
+            string senderSource,
+            ITestResultCollection<TestOperationResult> senderTestResults,
+            Option<string> receiverSource,
+            Option<ITestResultCollection<TestOperationResult>> receiverTestResults,
             string resultType,
             ITestResultComparer<TestOperationResult> testResultComparer,
             NetworkStatusTimeline networkStatusTimeline)
         {
+            if (receiverSource.HasValue && !receiverTestResults.HasValue)
```

Comment:
just to combine; and just throw argument exception with same message; e.g. "please provide both receiver source and receiver test results or neither".